### ISSUE TITLE
Add a lint for an undefined namespace

### ISSUE THREAD

**Issue by nex3** [kristerkari/stylelint-scss#513]
If an expression is written using a namespace (such as `foo.$bar` or `@include foo.bar`) and that namespace isn't actually provided by any `@use` rules in the file, that expression is guaranteed to be broken. It would be nice to add a lint to surface that breakage.

**Comment by kristerkari** [comment]
Sounds good, can you also provide some examples of the type of things that the rule should catch?

**Comment by nex3** [comment]
These would be invalid:

```scss
foo.$bar: value;
```

```scss
@debug foo.$bar;
```

```scss
@debug foo.bar();
```

```scss
@include foo.bar();
```

but they would all be valid if they also contained `@use "foo";` or `@use "qux" as foo;`.

**Comment by json-derulo** [comment]
I'm not sure if that's something a linter should cover. In my perspective, a linter is meant to be a tool which detects code smells and possible bugs in code, which **can** be compiled. But in this case, the compiler will always throw an error. So it seems redundant to me to add such a lint rule.

Instead, the IDE should display those errors inline. I know that WebStorm is capable of displaying a warning in this case.

**Comment by kristerkari** [comment]
@json-derulo I agree, it's not really worth to add a linting rule for something which the complier will catch anyway.
